Adds backup strategy section to readme (#185)
habitat-sh/on-prem-builder: README.md

```
This will spin up an Artifactory instance, and you can use the following to log in: http://localhost:8081/artifactory/webapp/#/home

### Backing up Builder Data

> The data that builder stores is luckily fairly lightweight and thus the backup and DR strategy is pretty straightforward. With On-Prem Builder we have two types of data that a user may want to restore in the case of a disaster. First, the package and user metadata, which gets stored in a PostgreSQL instance and second, habitat artifacts (.harts) which get stored in whichever artifact storage backend you've chosen to use (minio, s3, artifactory).

Reviewer: Minor - capitalize "Builder" here and in other references

Reviewer: Also capitalize "Habitat"

> The data that Builder stores is luckily fairly lightweight and thus the backup and DR strategy is pretty straightforward. With On-Prem Builder we have two types of data that a user may want to restore in the case of a disaster. First, the package and user metadata, which gets stored in a PostgreSQL instance and second, habitat artifacts (.harts) which get stored in whichever artifact storage backend you've chosen to use (Minio, S3, Artifactory).

Author: the data we store isn't heavyweight, meaning its not a massive database with a ton of data. its pretty small and pretty lightweight

> Ideally backups of the builder cluster would be performed in lock-step however, due to the nature of the data that builder stores there is very little concern with the timing of this operation. In the worst case if a package's metadata is missing from PostgreSQL, it can easily be repopulated by reuploading the package with the `--force` flag like so `hab pkg upload <path to hartfile> -u <on-prem_url> --force

Reviewer: "re-uploading" ?

> Ideally backups of the Builder cluster would be performed in lock-step however, due to the nature of the data that Builder stores there is very little concern with the timing of this operation. In the worst case if a package's metadata is missing from PostgreSQL, it can easily be repopulated by re-uploading the package with the `--force` flag like so `hab pkg upload <path to hartfile> -u <on-prem_url> --force`

Author: The backup(s) would be performed in lock-step. E.g. both of the activities we're outlining in this doc.

#### Database Backups

> The process of backing up Builder's PostgreSQL database should be no different than backing up any other PostgreSQL database. The process amounts to a pg_dump. If you have a backup strategy for othes production instances of PostgreSQL we suggest you use the same strategy here with the `builder` database. Otherwise the process of a manual backup would be as follows:

Reviewer: Typo: "othes"

> The process of backing up Builder's PostgreSQL database should be no different than backing up any other PostgreSQL database. The process amounts to a pg_dump. If you have a backup strategy for other production instances of PostgreSQL we suggest you use the same strategy here with the `builder` database. Otherwise the process of a manual backup would be as follows:

Reviewer 2: ```suggestion
Backing up Builder's PostgreSQL database is the same as backing up any PostgreSQL database. The process is a [pg_dump](https://www.postgresql.org/docs/11/app-pgdump.html). If you have a backup strategy for other production instances of PostgreSQL, then apply your backup pattern to the `builder` database. To backup your `builder` database manually, follow these steps:
```

1. Shut down the API to ensure no active transactions are occuring. (Optional but preferred)
        `hab svc stop habitat/builder-api`
1. Switch to user `hab`
        `sudo su - hab`
1. Find your Postgres password 
        `sudo cat /hab/svc/builder-api/config/config.toml`
1. Export as envvar 
        `export PGPASSWORD=<pw>`
1. Run pgdump 
        `/hab/pkgs/core/postgresql/<version>/<release>/bin/pg_dump --file=builder.dump --format=custom --host=<ip_of_pg_host> --dbname=builder`
1. Start the api and verify 
        `sudo hab svc start habitat/builder-api`

> Now the backup of your database has been dumped into the `builder.dump` file on your filesystem and can be moved or stored in whatever manner you see fit. For the backup to be useful in a DR scenario we obviously suggest moving the database dump somewhere else. An AWS S3 bucket would suffice for many but again we suggest you store the dump with the same strategy in which you store any other database dumps.

Reviewer 2: ```suggestion
Once the backup finishes,  your archive file will be named `builder.dump` file on your filesystem. Move and store this file according to your local policies. We recommend storing it remotely, either physically or virtually, so it will be useable in a worst-case scenario. For most, storing the dump file in an AWS bucket or Azure storage is enough, but you should follow the same strategy outlined in your disaster recovery plan for all database dumps. 
```

#### Database Restore

> Much like the backup process, there is no magic in the database restore for Builder. As such, if you already have a restoration strategy your organization is using, feel free to continue using for the `builder` database. Otherwise, a manual restore should be performed as follows:

Reviewer 2: ```suggestion
Restoring a `builder` database is exactly like restoring any other database--which is to say, there is no magical solution. If you already have a restoration strategy in place at your organization, follow that to restore your `builder` database.  To restore your  data `builder` database manually, follow these steps:
```

1. Switch to user `hab`
        `sudo su - hab`
1. Find your postgres password
        `sudo cat /hab/svc/builder-api/config/config.toml`
1. Export as envvar
        `export PGPASSWORD=<pw>`
1. Create the new builder database *
        `/hab/pkgs/core/postgresql/<version>/<release>/bin/createdb -w -h <url_of_pg_host> -p <configured_pg_port> -U hab builder`
1. Verify connectivity to the new database instance
        `/hab/pkgs/core/postgresql/<version>/<release>/bin/psql --host=<url_of_pg_host> --dbname=builder`
1. Restore the dump into the new DB
        `/hab/pkgs/core/postgresql/<version>/<release>/bin/pg_restore --host=<url_of_pg_host> --dbname=builder builder.dump`
1. Start the on-prem builder services

        * In some cases your version of postgres might not have a createdb binary in which case you'll want to connect to database to run the create db command

Just like that your database data should be restored and ready for new transactions!

#### Artifact Backups

> The process of artifact backups is quite a bit more environmentally subjective than postgres if only because we support more than one artifact storage backend. For the sake of these docs we will focus on minio backups.

Reviewer: Capitalize Postgres

Reviewer 2: What does "environmentally subjective" mean?

Author: It means that there is no single process because the process would be determined by the user environment.
